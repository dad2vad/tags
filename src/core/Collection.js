class Collection {
  constructor(data = []) {
    this.data = data;
    this.count = data.length;
  }

  /***
   * Check if Collection has give item.
   *
   * @param item
   * @returns {boolean}
   */
  has(item) {
    return this.data.indexOf(item) > -1;
  }

  /**
   * Add unique item to Collection.
   *
   * @param item
   * @returns {Collection}
   */
  add(item) {
    if (!this.has(item)) {
      this.count = this.count + 1;
      this.data.push(item);
    }
    return this;
  }

  /***
   * Remove item from Collection.
   *
   * @param item
   * @returns {Collection}
   */
  remove(item) {
    if (this.has(item)) {
      this.data.splice(this.data.indexOf(item), 1);
      this.count = this.count - 1;
    }

    return this;
  }

  /***
   * Remove first item from Collection.
   *
   * @returns {Collection}
   */
  removeFirst() {
    if (this.count > 0) {
      this.data.shift();
      this.count = this.count - 1;
    }

    return this;
  }

  /***
   * Return total size of Collection items.
   *
   * @returns {*}
   */
  size() {
    return this.count;
  }

  /**
   * Return all elements from Collection.
   *
   * @returns {Array|*}
   */
  all() {
    return this.data;
  }
}

export default Collection;
