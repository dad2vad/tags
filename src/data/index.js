import { AmazeUI } from './frameworks/AmazeUI';
import { Axiom } from './frameworks/Axiom';
import { BlazeCSS } from './frameworks/BlazeCSS';
import { Bootstrap3, Bootstrap4 } from './frameworks/Bootstrap';
import { Bulma } from './frameworks/Bulma';
import { Foundation5, Foundation6 } from './frameworks/Foundation';
import { Gumby } from './frameworks/Gumby';
import { Ink } from './frameworks/Ink';
import { Kube } from './frameworks/Kube';
import { Materialize } from './frameworks/Materialize';
import { MetroUI } from './frameworks/MetroUI';
import { Mini } from './frameworks/Mini';
import { MUI } from './frameworks/MUI';
import { OfficeUI } from './frameworks/OfficeUI';
import { UIkit2, UIkit3 } from './frameworks/UIkit';
import { Pure } from './frameworks/Pure';
import { Primer } from './frameworks/Primer';
import { SemanticUI } from './frameworks/SemanticUI';
import { Skeleton } from './frameworks/Skeleton';
import { Spectre } from './frameworks/Spectre';

import Accordion from './components/Accordion';
import Alert from './components/Alert';
import Avatar from './components/Avatar';
import Badge from './components/Badge';
import Breadcrumbs from './components/Breadcrumbs';
import Button from './components/Button';
import Card from './components/Card';
import Dropdown from './components/Dropdown';
import FlexGrid from './components/FlexGrid';
import Form from './components/Form';
import Grid from './components/Grid';
import Icons from './components/Icons';
import Label from './components/Label';
import List from './components/List';
import Loading from './components/Loading';
import Media from './components/Media';
import Menu from './components/Menu';
import Modal from './components/Modal';
import Navbar from './components/Navbar';
import OffCanvas from './components/OffCanvas';
import Pagination from './components/Pagination';
import Panel from './components/Panel';
import Popover from './components/Popover';
import Progress from './components/Progress';
import Table from './components/Table';
import Tabs from './components/Tabs';
import Tags from './components/Tags';
import Thumbnail from './components/Thumbnail';
import Tooltip from './components/Tooltip';
import Typography from './components/Typography';
import Utilities from './components/Utilities';

export default {
  frameworks: [
    AmazeUI,
    Axiom,
    BlazeCSS,
    Bootstrap3,
    Bootstrap4,
    Bulma,
    Foundation5,
    Foundation6,
    Gumby,
    Ink,
    Kube,
    Materialize,
    MetroUI,
    Mini,
    MUI,
    OfficeUI,
    UIkit2,
    UIkit3,
    Pure,
    Primer,
    SemanticUI,
    Skeleton,
    Spectre,
  ],
  components: [
    Accordion,
    Alert,
    Avatar,
    Badge,
    Breadcrumbs,
    Button,
    Card,
    Dropdown,
    FlexGrid,
    Form,
    Grid,
    Icons,
    Label,
    List,
    Loading,
    Media,
    Menu,
    Modal,
    Navbar,
    OffCanvas,
    Pagination,
    Panel,
    Popover,
    Progress,
    Table,
    Tabs,
    Tags,
    Thumbnail,
    Tooltip,
    Typography,
    Utilities,
  ],
};
