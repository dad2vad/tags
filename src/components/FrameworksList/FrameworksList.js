import React from 'react';
import { array, instanceOf } from 'prop-types';
import Icon from '../Icon';
import Link from '../Link';
import { toInstall } from '../../core/utils';
import Collection from '../../core/Collection';

const FrameworksList = ({ frameworks, visible }) => {
  const renderNoBadge = () => <span className="tag is-rounded is-small is-danger">No</span>;

  const renderHeader = () => {
    return (
      <tr>
        <th className="is-first" />
        {frameworks.filter(framework => visible.has(framework.slug)).map(framework => {
          return (
            <th key={framework.slug}>
              <span data-tooltip={framework.description} className="tag is-tooltip is-info">
                {framework.name}
              </span>
            </th>
          );
        })}
      </tr>
    );
  };

  const renderLinksRow = () => {
    return (
      <tr>
        <td>
          <strong>Links</strong>
        </td>
        {frameworks.filter(framework => visible.has(framework.slug)).map(framework => {
          return (
            <td key={framework.slug}>
              <div className="field has-addons has-text-centered">
                <div className="control">
                  <Link url={framework.url} className="button is-small is-info is-outlined">
                    <Icon name="globe" />
                    <span className="button-text">Website</span>
                  </Link>
                </div>
                <div className="control">
                  <Link
                    url={framework.documentation}
                    className="button is-small is-info is-outlined"
                  >
                    <Icon name="book" />
                    <span className="button-text">Docs</span>
                  </Link>
                </div>
                <div className="control">
                  <Link url={framework.github} className="button is-small is-info is-outlined">
                    <Icon name="github" />
                    <span className="button-text">GitHub</span>
                  </Link>
                </div>
              </div>
            </td>
          );
        })}
      </tr>
    );
  };

  const renderInstallRow = () => {
    return (
      <tr>
        <td>
          <strong>How To Install?</strong>
        </td>
        {frameworks.filter(framework => visible.has(framework.slug)).map(framework => {
          return (
            <td className="no-padding" key={framework.slug}>
              <pre>
                <code>{toInstall(framework, 'bower')}</code>
              </pre>
              <pre>
                <code>{toInstall(framework, 'npm')}</code>
              </pre>
              <pre>
                <code>{toInstall(framework, 'yarn')}</code>
              </pre>
            </td>
          );
        })}
      </tr>
    );
  };

  const renderStatsRow = () => {
    return (
      <tr>
        <td>
          <strong>Github Stats</strong>
        </td>
        {frameworks.filter(framework => visible.has(framework.slug)).map(framework => {
          return (
            <td key={framework.slug}>
              <div className="tag-group">
                <Link
                  url={`${framework.github}/stargazers`}
                  className="tag is-space is-rounded is-small is-dark"
                >
                  <Icon name="star" />
                  {framework.stats.stars}
                </Link>
                <Link
                  url={`${framework.github}/network`}
                  className="tag is-rounded is-small is-dark"
                >
                  <Icon name="code-fork" />
                  {framework.stats.forks}
                </Link>
              </div>
            </td>
          );
        })}
      </tr>
    );
  };

  const renderVersionRow = () => {
    return (
      <tr>
        <td>
          <strong>Latest Version</strong>
        </td>
        {frameworks.filter(framework => visible.has(framework.slug)).map(framework => {
          return (
            <td key={framework.slug}>
              <Link
                url={`${framework.github}/releases`}
                className="tag is-space is-rounded is-small is-info"
              >
                <Icon name="history" />
                <span>v{framework.version}</span>
              </Link>
            </td>
          );
        })}
      </tr>
    );
  };

  const renderIconsRow = () => {
    return (
      <tr>
        <td>
          <strong>Icons</strong>
        </td>
        {frameworks.filter(framework => visible.has(framework.slug)).map(framework => {
          return (
            <td key={framework.slug}>
              {framework.icons && (
                <Link
                  url={framework.icons.url}
                  tooltip={framework.icons.name}
                  className="tag is-rounded is-small is-success"
                  text="Yes"
                />
              )}
              {!framework.icons && <span className="tag is-rounded is-small is-danger">No</span>}
            </td>
          );
        })}
      </tr>
    );
  };

  const renderCSSRow = () => {
    return (
      <tr>
        <td>
          <strong>CSS Preprocessor</strong>
        </td>
        {frameworks.filter(framework => visible.has(framework.slug)).map(framework => {
          return (
            <td key={framework.slug}>
              {framework.css && (
                <Link
                  url={framework.css === 'LESS' ? 'http://lesscss.org/' : 'http://sass-lang.com/'}
                  className="tag is-rounded is-small is-info"
                >
                  {framework.css}
                </Link>
              )}
              {!framework.css && renderNoBadge()}
            </td>
          );
        })}
      </tr>
    );
  };

  const renderJSRow = () => {
    return (
      <tr>
        <td>
          <strong>JS Library</strong>
        </td>
        {frameworks.filter(framework => visible.has(framework.slug)).map(framework => {
          return (
            <td key={framework.slug}>
              {framework.js && (
                <Link
                  url={framework.js}
                  className="tag is-rounded is-small is-success"
                  text="Yes"
                />
              )}
              {!framework.js && renderNoBadge()}
            </td>
          );
        })}
      </tr>
    );
  };

  const renderLicenceRow = () => {
    return (
      <tr>
        <td>
          <strong>License</strong>
        </td>
        {frameworks.filter(framework => visible.has(framework.slug)).map(framework => {
          return (
            <td key={framework.slug}>
              {framework.licence && (
                <Link
                  url={framework.licence.url}
                  className="tag is-rounded is-small is-info"
                  text={framework.licence.name}
                />
              )}
              {!framework.licence && (
                <Link
                  url="https://opensource.org/licenses/MIT"
                  className="tag is-rounded is-small is-info"
                  text="MIT"
                />
              )}
            </td>
          );
        })}
      </tr>
    );
  };

  return (
    <div className="section is-x">
      <div className="container is-fullhd">
        <h2 className="title">List of Frameworks</h2>
        <table className="table has-text-centered is-striped is-bordered">
          <thead>{renderHeader()}</thead>
          <tbody>
            {renderLinksRow()}
            {renderInstallRow()}
            {renderStatsRow()}
            {renderVersionRow()}
            {renderIconsRow()}
            {renderCSSRow()}
            {renderJSRow()}
            {renderLicenceRow()}
          </tbody>
        </table>
      </div>
    </div>
  );
};

FrameworksList.propTypes = {
  frameworks: array.isRequired,
  visible: instanceOf(Collection).isRequired,
};

export default FrameworksList;
