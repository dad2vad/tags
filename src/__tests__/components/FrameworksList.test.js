import React from 'react';
import { shallow } from 'enzyme';
import Collection from '../../core/Collection';
import data from '../../data';
import FrameworksList from '../../components/FrameworksList';

describe('Frameworks List', () => {
  it('should pass snapshot correctly', () => {
    const visible = new Collection(['bulma', 'ink']);
    const component = shallow(<FrameworksList frameworks={data.frameworks} visible={visible} />);
    expect(component).toMatchSnapshot();
  });
});
