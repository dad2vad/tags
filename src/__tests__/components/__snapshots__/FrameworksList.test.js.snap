// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frameworks List should pass snapshot correctly 1`] = `
<div
  className="section is-x"
>
  <div
    className="container is-fullhd"
  >
    <h2
      className="title"
    >
      List of Frameworks
    </h2>
    <table
      className="table has-text-centered is-striped is-bordered"
    >
      <thead>
        <tr>
          <th
            className="is-first"
          />
          <th>
            <span
              className="tag is-tooltip is-info"
              data-tooltip="A modern CSS framework based on Flexbox"
            >
              Bulma
            </span>
          </th>
          <th>
            <span
              className="tag is-tooltip is-info"
              data-tooltip="Ink is a set of tools for quick development of web interfaces"
            >
              Ink Interface Kit
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <strong>
              Links
            </strong>
          </td>
          <td>
            <div
              className="field has-addons has-text-centered"
            >
              <div
                className="control"
              >
                <Link
                  className="button is-small is-info is-outlined"
                  url="http://bulma.io/"
                >
                  <Icon
                    name="globe"
                  />
                  <span
                    className="button-text"
                  >
                    Website
                  </span>
                </Link>
              </div>
              <div
                className="control"
              >
                <Link
                  className="button is-small is-info is-outlined"
                  url="http://bulma.io/documentation/overview/start/"
                >
                  <Icon
                    name="book"
                  />
                  <span
                    className="button-text"
                  >
                    Docs
                  </span>
                </Link>
              </div>
              <div
                className="control"
              >
                <Link
                  className="button is-small is-info is-outlined"
                  url="https://github.com/jgthms/bulma"
                >
                  <Icon
                    name="github"
                  />
                  <span
                    className="button-text"
                  >
                    GitHub
                  </span>
                </Link>
              </div>
            </div>
          </td>
          <td>
            <div
              className="field has-addons has-text-centered"
            >
              <div
                className="control"
              >
                <Link
                  className="button is-small is-info is-outlined"
                  url="http://ink.sapo.pt/"
                >
                  <Icon
                    name="globe"
                  />
                  <span
                    className="button-text"
                  >
                    Website
                  </span>
                </Link>
              </div>
              <div
                className="control"
              >
                <Link
                  className="button is-small is-info is-outlined"
                  url="http://ink.sapo.pt/ui-elements/"
                >
                  <Icon
                    name="book"
                  />
                  <span
                    className="button-text"
                  >
                    Docs
                  </span>
                </Link>
              </div>
              <div
                className="control"
              >
                <Link
                  className="button is-small is-info is-outlined"
                  url="https://github.com/sapo/Ink/"
                >
                  <Icon
                    name="github"
                  />
                  <span
                    className="button-text"
                  >
                    GitHub
                  </span>
                </Link>
              </div>
            </div>
          </td>
        </tr>
        <tr>
          <td>
            <strong>
              How To Install?
            </strong>
          </td>
          <td
            className="no-padding"
          >
            <pre>
              <code>
                bower i bulma -S
              </code>
            </pre>
            <pre>
              <code>
                npm i bulma -S
              </code>
            </pre>
            <pre>
              <code>
                yarn add bulma
              </code>
            </pre>
          </td>
          <td
            className="no-padding"
          >
            <pre>
              <code>
                bower i sapo-ink -S
              </code>
            </pre>
            <pre>
              <code>
                Not Available
              </code>
            </pre>
            <pre>
              <code>
                Not Available
              </code>
            </pre>
          </td>
        </tr>
        <tr>
          <td>
            <strong>
              Github Stats
            </strong>
          </td>
          <td>
            <div
              className="tag-group"
            >
              <Link
                className="tag is-space is-rounded is-small is-dark"
                url="https://github.com/jgthms/bulma/stargazers"
              >
                <Icon
                  name="star"
                />
                19K+
              </Link>
              <Link
                className="tag is-rounded is-small is-dark"
                url="https://github.com/jgthms/bulma/network"
              >
                <Icon
                  name="code-fork"
                />
                1.4K+
              </Link>
            </div>
          </td>
          <td>
            <div
              className="tag-group"
            >
              <Link
                className="tag is-space is-rounded is-small is-dark"
                url="https://github.com/sapo/Ink//stargazers"
              >
                <Icon
                  name="star"
                />
                1.9K+
              </Link>
              <Link
                className="tag is-rounded is-small is-dark"
                url="https://github.com/sapo/Ink//network"
              >
                <Icon
                  name="code-fork"
                />
                200+
              </Link>
            </div>
          </td>
        </tr>
        <tr>
          <td>
            <strong>
              Latest Version
            </strong>
          </td>
          <td>
            <Link
              className="tag is-space is-rounded is-small is-info"
              url="https://github.com/jgthms/bulma/releases"
            >
              <Icon
                name="history"
              />
              <span>
                v
                0.5.3
              </span>
            </Link>
          </td>
          <td>
            <Link
              className="tag is-space is-rounded is-small is-info"
              url="https://github.com/sapo/Ink//releases"
            >
              <Icon
                name="history"
              />
              <span>
                v
                3.1.10
              </span>
            </Link>
          </td>
        </tr>
        <tr>
          <td>
            <strong>
              Icons
            </strong>
          </td>
          <td>
            <Link
              className="tag is-rounded is-small is-success"
              text="Yes"
              tooltip="Font Awesome (Optional)"
              url="http://fontawesome.io/"
            />
          </td>
          <td>
            <span
              className="tag is-rounded is-small is-danger"
            >
              No
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <strong>
              CSS Preprocessor
            </strong>
          </td>
          <td>
            <Link
              className="tag is-rounded is-small is-info"
              url="http://sass-lang.com/"
            >
              SASS
            </Link>
          </td>
          <td>
            <Link
              className="tag is-rounded is-small is-info"
              url="http://sass-lang.com/"
            >
              SASS
            </Link>
          </td>
        </tr>
        <tr>
          <td>
            <strong>
              JS Library
            </strong>
          </td>
          <td>
            <span
              className="tag is-rounded is-small is-danger"
            >
              No
            </span>
          </td>
          <td>
            <Link
              className="tag is-rounded is-small is-success"
              text="Yes"
              url="http://ink.sapo.pt/javascript/"
            />
          </td>
        </tr>
        <tr>
          <td>
            <strong>
              License
            </strong>
          </td>
          <td>
            <Link
              className="tag is-rounded is-small is-info"
              text="MIT"
              url="https://opensource.org/licenses/MIT"
            />
          </td>
          <td>
            <Link
              className="tag is-rounded is-small is-info"
              text="MIT"
              url="https://opensource.org/licenses/MIT"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
